eZ/Publish/API/Repository/Tests/Iterator/BatchIteratorAdapter/AbstractSearchAdapterTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository\Tests\Iterator\BatchIteratorAdapter;
+
+use eZ\Publish\API\Repository\Iterator\BatchIteratorAdapter\AbstractSearchAdapter;
+use eZ\Publish\API\Repository\SearchService;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\MatchAll;
+use eZ\Publish\API\Repository\Values\Content\Search\SearchResult;
+use Iterator;
+use PHPUnit\Framework\TestCase;
+
+abstract class AbstractSearchAdapterTest extends TestCase
+{
+    protected const EXAMPLE_LANGUAGE_FILTER = [
+        'languages' => ['eng-GB', 'pol-PL'],
+        'useAlwaysAvailable' => true,
+    ];
+
+    protected const EXAMPLE_OFFSET = 7;
+    protected const EXAMPLE_LIMIT = 13;
+
+    final public function testFetch(): void
+    {
+        $expectedIterator = $this->createMock(Iterator::class);
+
+        $searchResults = $this->createMock(SearchResult::class);
+        $searchResults->method('getIterator')->willReturn($expectedIterator);
+
+        $originalQuery = $this->newQuery();
+        $originalQuery->filter = new MatchAll();
+
+        $expectedQuery = $this->newQuery();
+        $expectedQuery->filter = new MatchAll();
+        $expectedQuery->offset = self::EXAMPLE_OFFSET;
+        $expectedQuery->limit = self::EXAMPLE_LIMIT;
+
+        $searchService = $this->createMock(SearchService::class);
+        $searchService
+            ->expects($this->once())
+            ->method($this->getExpectedFindMethod())
+            ->with($expectedQuery, self::EXAMPLE_LANGUAGE_FILTER, true)
+            ->willReturn($searchResults);
+
+        $adapter = $this->createAdapterUnderTest($searchService, $originalQuery, self::EXAMPLE_LANGUAGE_FILTER, true);
+
+        $this->assertEquals(
```

Comment:
I'm almost certain that this will be the same `Iterator` instance, so we can use `assertSame` in this case (and it makes more sense since you're comparing an empty, mocked iterator).

Reply:
Fixed in https://github.com/ezsystems/ezplatform-kernel/pull/211/commits/dec2587157cb1b24fd7ebd111daa9b80bd7145e8